src/ImageSharp/Compression/Zlib/DeflaterEngine.cs

```diff
@@ -486,15 +486,16 @@ private bool FindLongestMatch(int curMatch)
             this.matchLen = Math.Max(this.matchLen, DeflaterConstants.MIN_MATCH - 1);
             int matchLength = this.matchLen;
```

Comment:
```suggestion
            int matchLength = this.matchLen;
            matchLength  = Math.Max(matchLength, DeflaterConstants.MIN_MATCH - 1);
            this.matchLen = matchLength;
```
Should give better codegen -- especially for older runtimes (JIT).

Reply:
It looks strange and somehow duplicate in code. So if it helps the JIT codegen, I agree to use it.